### PR TITLE
* Improvement: Better options for delete/edit/duplicate action of each item in Price list block.

### PR DESCRIPTION
**Pull request by AbhijitNage123**
### Description

Trello - https://trello.com/c/xtRTLSA5/68-backward-compatibility-price-list

### Types of changes

Breaking change

### How has this been tested?

- Created Price list in the master [ Add image, Title, Price, and Description ].
- Switch to the `backward_comp_price_list` branch and check is there anything is broken or not.

### Checklist:
- [x] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

**Comment by vrundakansara** [comment]
- Create a Price List in master branch.
- Checkout to this branch.
- Refresh the frontend page.

You will see the issue.
Issue - https://share.getcloudapp.com/xQuAXBwD
Expected output - https://share.getcloudapp.com/YEuy97DX

**Comment by vrundakansara** [comment]
- Image position not working - https://share.getcloudapp.com/E0ur1wNo

**Comment by AbhijitNage123** [comment]
> * Create a Price List in master branch.
> * Checkout to this branch.
> * Refresh the frontend page.
> 
> You will see the issue.
> Issue - https://share.getcloudapp.com/xQuAXBwD
> Expected output - https://share.getcloudapp.com/YEuy97DX

Fix and tested working fine.

**Comment by AbhijitNage123** [comment]
> * Image position not working - https://share.getcloudapp.com/E0ur1wNo
> * Adding a new price list throws an error - https://share.getcloudapp.com/E0ur1wvv

Image position issue - Done
Adding new price list throws an error - Done & tested working fine

File: src/blocks/price-list-child/components/Description.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+const {
+	RichText,
+} = wp.blockEditor
+
+const { __ } = wp.i18n
+
+const {
+	createBlock
```
Comment (RajkiranBagal) [inline src/blocks/price-list-child/components/Description.js:8]: Remove unused components.

File: src/blocks/price-list-child/components/Price.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
+const {
+	RichText,
+} = wp.blockEditor
+
+const { __ } = wp.i18n
+
+const {
+	createBlock
```
Comment (RajkiranBagal) [inline src/blocks/price-list-child/components/Price.js:8]: Remove unused components.

File: src/blocks/price-list-child/components/Title.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
+const {
+	RichText,
+} = wp.blockEditor
+
+const { __ } = wp.i18n
+const {
+	createBlock
```
Comment (RajkiranBagal) [inline src/blocks/price-list-child/components/Title.js:7]: Remove unused components.

File: src/blocks/price-list-child/edit.js [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+
+class UAGBRestaurantMenuChild extends Component {
+
+    constructor() {
+
+		super( ...arguments )
+		this.onSelectRestImage  = this.onSelectRestImage.bind( this )
+		this.onRemoveRestImage  = this.onRemoveRestImage.bind(this)
+		this.getImageName       = this.getImageName.bind(this)
+		// this.onRemoveImage 		= this.onRemoveImage.bind( this )
```
Comment (RajkiranBagal) [inline src/blocks/price-list-child/edit.js:48]: Removed commented code

File: src/blocks/price-list-child/inline-styles.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+		
+	} = props.attributes
+
+
+	var selectors = {
+		
+	
+	}
+
+
```
Comment (RajkiranBagal) [inline src/blocks/price-list-child/inline-styles.js:24]: Remove the file if this is not needed

File: src/blocks/price-list-child/save.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+				) }
+				>
+			<div className={ classnames(
+					className,
+					"uagb-rest_menu__outer-wrap",
+					`uagb-block-${block_id}`
+				) }
+				>
+						<div className = { classnames(
+							// "uagb-rest_menu__wrap",
```
Comment (RajkiranBagal) [inline src/blocks/price-list-child/save.js:44]: Remove commented code.

File: src/blocks/price-list-child/style.scss [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+//         display: block;
+//         width: 100%;
+//         float: left;
+//         padding-left: 10px;
+//         padding-right: 10px;
+//     }
+
+// @media(max-width:976px) {
+
+
```
Comment (RajkiranBagal) [inline src/blocks/price-list-child/style.scss:32]: Remove the unwanted commented code

File: src/blocks/price-list-child/attributes.js [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+		type: "boolean",
+		default: false
+	},
+	description:{
+		type: "string",
+		default: "Lorem ipsum dolor sit amet, consectetur adipiscing elit."
+	},
+	title:{
+		type: "string",
+		default:  "Menu Item"
```
Comment (RajkiranBagal) [inline src/blocks/price-list-child/attributes.js:18]: Make this Translatable.

File: src/blocks/price-list/edit.js [hunk 9]
```diff
@@ -22,148 +20,161 @@ import WebfontLoader from "../../components/typography/fontloader"
...
 const {
+	InspectorControls,
+	PanelColorSettings,
 	AlignmentToolbar,
 	BlockControls,
 	ColorPalette,
-	InspectorControls,
-	PanelColorSettings,
-	MediaUpload
+	MediaUpload,
```
Comment (RajkiranBagal) [inline src/blocks/price-list/edit.js:28]: Remove all used components.

File: src/blocks/price-list/edit.js [hunk 10]
```diff
@@ -22,148 +20,161 @@ import WebfontLoader from "../../components/typography/fontloader"
...
+
+const { select, withSelect } = wp.data;
+
 const { Component, Fragment } = wp.element
 
 const MAX_COLUMNS = 3
 
-class UAGBRestaurantMenu extends Component {
+const ALLOWED_BLOCKS = [ "uagb/restaurant-menu-child" ]
+// const ALLOWED_BLOCKS = [ "core/buttons" ]
```
Comment (RajkiranBagal) [inline src/blocks/price-list/edit.js:53]: Remove commented code.

File: src/blocks/price-list/edit.js [hunk 11]
```diff
@@ -22,148 +20,161 @@ import WebfontLoader from "../../components/typography/fontloader"
...
+		
 		super( ...arguments )
-		this.onSelectRestImage  = this.onSelectRestImage.bind( this )
-		this.onRemoveRestImage  = this.onRemoveRestImage.bind(this)
-		this.getImageName       = this.getImageName.bind(this)
+		
 		this.onRemoveImage 		= this.onRemoveImage.bind( this )
 		this.onSelectImage 		= this.onSelectImage.bind( this )
+		this.setcolumns			= this.setcolumns.bind( this )
+		this.setheadingTag			= this.setheadingTag.bind( this )
```
Comment (RajkiranBagal) [inline src/blocks/price-list/edit.js:65]: Indent the code

Comment (AbhijitNage123) [reply src/blocks/price-list/edit.js]: Done

File: src/blocks/price-list/edit.js (hunk 10 above)
Comment (AbhijitNage123) [reply src/blocks/price-list/edit.js]: Removed

File: src/blocks/price-list/edit.js (hunk 9 above)
Comment (AbhijitNage123) [reply src/blocks/price-list/edit.js]: Removed

File: src/blocks/price-list-child/attributes.js (hunk 8 above)
Comment (AbhijitNage123) [reply src/blocks/price-list-child/attributes.js]: Done

**Review by RajkiranBagal — APPROVED** [review]

**Review by vrundakansara — APPROVED** [review]

**Pull request merged**